### PR TITLE
Epmlstrtai 803

### PR DESCRIPTION
**Pull request by stelmakhmaksim**
Made user login and registration of new user.

### CODE REVIEW

File: jdi-cucumber-test-generator-integration-tests/src/test/java/com/epam/test_generator/api/steps/when/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import static com.epam.test_generator.api.BddGeneratorApi.loginUsingPOST;
+
+import com.epam.http.response.RestResponse;
+import com.epam.test_generator.api.steps.StepBackground;
+import com.epam.test_generator.controllers.user.request.LoginUserDTO;
+import cucumber.api.java.en.When;
+
+public class Authentication extends StepBackground {
+
+    @When("^I want to sign in$")
```
Comment (pomkine) [inline jdi-cucumber-test-generator-integration-tests/src/test/java/com/epam/test_generator/api/steps/when/Authentication.java]: "I want" is not an action. "I sign in" is better

File: src/main/java/com/epam/test_generator/services/EmailServiceProd.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+
+@Transactional
+@PropertySource("classpath:email.messages.properties")
+@Profile("!integration-tests")
+public class EmailServiceProd implements EmailService {
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/services/EmailServiceProd.java]: Please think about better naming for this class

Comment (stelmakhmaksim) [reply src/main/java/com/epam/test_generator/services/EmailServiceProd.java]: Fixed

File: jdi-cucumber-test-generator-integration-tests/src/test/java/com/epam/test_generator/api/steps/when/Authentication.java (hunk 1 above)
Comment (stelmakhmaksim) [reply jdi-cucumber-test-generator-integration-tests/src/test/java/com/epam/test_generator/api/steps/when/Authentication.java]: Fixed

**Pull request merged**